metadata/src/utils/validation.rs

```diff
@@ -195,11 +212,38 @@ pub fn get_type_hash(
 fn get_extrinsic_hash(
     registry: &PortableRegistry,
     extrinsic: &ExtrinsicMetadata,
+    specific_pallets: &Option<Vec<&str>>,
 ) -> [u8; HASH_LEN] {
     let mut visited_ids = HashSet::<u32>::new();
 
-    let mut bytes = concat_and_hash2(
-        &get_type_hash(registry, extrinsic.ty, &mut visited_ids),
+    // Get the hashes of the extrinsic type.
+    let address_hash = get_type_hash(registry, extrinsic.address_ty, &mut visited_ids);
+
+    // Get the hash of the `RuntimeCall` for the specified pallets.
+    // This mirrors the metadata stripped by `retain_metadata()`.
+    let call_hash = {
+        let ty = registry
+            .types
+            .get(extrinsic.call_ty as usize)
+            .expect("Metadata should contain call enum type in registry");
+
+        // If the `RuntimeCall` is a variant, then filter out the pallets of interest.
+        // Note: this can be a different type def for our testing.
+        if let TypeDef::Variant(variant) = &ty.ty.type_def {
+            get_type_def_variant_hash(registry, variant, specific_pallets, &mut visited_ids)
+        } else {
+            get_type_hash(registry, extrinsic.signature_ty, &mut visited_ids)
```

Comment:
Why `extrinsic.signature_ty` here; is this supposed to just be some arbitrary hash that's returned to make testing a little easier?

If so, maybe just return some arbitrary thing, orr call `get_type_hash` on the entire `ty` if it's not a variant for "correctness"?

Reply:
Thanks for spotting this! I mean to have `call_ty` here instead of `signature_ty`.
If the `RuntimeCall` is an enum (as should be defined by substrate), then we know which variants of the enum to include in the hash. Otherwise, we hash everything from that type (since we may add a struct or a tuple in the metadata for our testing).

Reply:
Makes sense to me :)